### PR TITLE
[Diffusion] Replace Conv3d with reshape + F.linear in PatchEmbed

### PR DESCRIPTION
**Pull request by AichenF**
## Motivation

For 5D video inputs, PatchEmbed uses nn.Conv3d with kernel_size == stride (non-overlapping patches). Under the hood, PyTorch's Conv3d emits three separate CUDA kernels:

- aten::fill — zeros the output buffer
- aten::copy — performs im2col rearrangement
- GEMM — the actual matrix multiply

Since the patches are non-overlapping, the convolution is mathematically equivalent to reshaping the input into patch-sized token rows and applying a linear projection. This PR replaces Conv3d with reshape + permute + F.linear, collapsing the operation into a single GEMM and eliminating the unnecessary fill and im2col overhead.

## Modifications

python/sglang/multimodal_gen/runtime/layers/visual_embedding.py (1 file, +24 / -1)
Added an optimized path for 5D inputs that reshapes [B, C, T, H, W] into [B, num_patches, patch_volume] and applies F.linear with a zero-copy view of the existing Conv3d weight
The original Conv3d path is kept as a fallback for non-5D inputs

## Benchmarking and Profiling
`sglang generate --model-path /Wan-AI/Wan2.1-I2V-14B-720P-Diffusers     --text-encoder-cpu-offload --pin-cpu-memory    --image-path /a_curious_raccoon.png --prompt "A curious raccoon"   --height 480  --width 832  --num-frames 41   --save-output      --num-gpus 8     --ulysses-degree 8     --ring-degree 1`
H200：
| Frames | Baseline (s/step) | Optimized (s/step) | Speedup |
|:------:|:-----------------:|:-----------------:|:-------:|
| 21 | 2.0552 | 1.7216 | **+16.2%** |
| 41 | 2.4155 | 2.0346 | **+15.8%** |
| 81 | 2.9553 | 2.9301 | +0.8% |
| 101 | 4.1475 | 4.1136 | +0.8% |

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/layers/visual_embedding.py [hunk 1]
```diff
@@ -79,9 +81,30 @@ def __init__(
...
+        if x.dim() == 5:
+            B, C, T, H, W = x.shape
+            pt, ph, pw = self.patch_size
+            T_ = T // pt
+            H_ = H // ph
+            W_ = W // pw
+
+            x = x.reshape(B, C, T_, pt, H_, ph, W_, pw)
+            x = x.permute(0, 2, 4, 6, 1, 3, 5, 7).contiguous()
+            x = x.reshape(B, T_ * H_ * W_, C * pt * ph * pw)
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/layers/visual_embedding.py]: we should at least allow fallback when the shape of x isn't dividable with patch_size

**Review by mickqian — COMMENTED** [review]
we should conduct more tests on accuracy and performance

File: python/sglang/multimodal_gen/test/unit/manual/bench_patch_embed.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
+"""
```
Comment (mickqian) [inline python/sglang/multimodal_gen/test/unit/manual/bench_patch_embed.py:1]: could you add a `manual` folder under `unit`?

Comment (AichenF) [reply python/sglang/multimodal_gen/test/unit/manual/bench_patch_embed.py:1]: sure

**Review by mickqian — APPROVED** [review]
